salt/_modules/solutions.py

```diff
@@ -0,0 +1,115 @@
+'''
+Various utilities to handle solution.
+'''
+
+import re
+import logging
+import time
+import os
+import yaml
+
+import salt.utils.json
+from salt.exceptions import CommandExecutionError
+
+HAS_LIBS = True
+SOLUTIONS_CONFIG_MAP = 'solutions'
+SOLUTIONS_CONFIG_MAP_NAMESPACE = 'default'
+SOLUTIONS_CONFIG_FILE = '/etc/metalk8s/solutions.yaml'
+try:
+    import kubernetes.client
+    from kubernetes.client.rest import ApiException
+    from pprint import pprint
+except ImportError:
+    HAS_LIBS = False
+
+log = logging.getLogger(__name__)
+
+
+__virtualname__ = 'solutions'
```

Comment:
Not really like `solutions` if one day salt create a module named solutions that will override it. IMO you should prefix your module with `metalk8s`